lib/ansible/cli/vault.py

```diff
@@ -318,6 +318,9 @@ def execute_encrypt_string(self):
             if stdin_text == '':
                 raise AnsibleOptionsError('stdin was empty, not encrypting')
 
+            if not sys.stdout.isatty() and stdin_text.endswith("\n"):
+                display.display("Warning: Provided string ends with a newline. It is unlikely this is correct. Did you mean 'echo -n'?", stderr=True)
```

Comment:
use `display.warning` for warnings

Reply:
Updated to warnings, but I wasn't able to figure out how to detect a warning from the current vault tests.